patches/sitecustomize.py

```diff
@@ -73,3 +75,29 @@ def exec_module(self, module):
 
 if not hasattr(sys, 'frozen'):
     sys.meta_path.insert(0, GcpModuleFinder())
+
+@wrapt.when_imported('google.generativeai')
+def post_import_logic(module):
+    if os.getenv('KAGGLE_DISABLE_GOOGLE_GENERATIVE_AI_INTEGRATION') == 1:
+      return
+    if os.getenv('KAGGLE_DATA_PROXY_TOKEN') == 0 || os.getenv('KAGGLE_USER_SECRETS_TOKEN') == 0:
+      return
+    old_configure = module.configure
+    def new_configure(*args, **kwargs):
+        if ('default_metadata' in kwargs):
+            default_metadata = kwargs['default_metadata']
+        else:
+            default_metadata = []
+        kwargs['transport'] = 'rest' # Only support REST requests for now
+        default_metadata.append(("x-kaggle-proxy-data", os.environ['KAGGLE_DATA_PROXY_TOKEN']))
+        default_metadata.append(('x-kaggle-authorization', f'Bearer {os.environ['KAGGLE_USER_SECRETS_TOKEN']}'))
+        kwargs['default_metadata'] = default_metadata
+        if ('client_options' in kwargs):
+            client_options = kwargs['client_options']
+        else:
+            client_options = {}
+        client_options['api_endpoint'] = os.environ['KAGGLE_DATA_PROXY_URL'] + '/palmapi'
```

Comment:
Same: hard coding this endpoint here means you will never be able to change it without breaking all the users using this image